ipn/localapi/localapi.go

```diff
@@ -1368,6 +1368,12 @@ func (h *Handler) servePrefs(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
+		if err := h.b.MaybeClearAppConnector(mp); err != nil {
```

Comment:
It may be better to move the error writing stuff into somewhere common for both paths. One option is to set `err` here and then skip setting `prefs` when `err != nil`. Another is to move the error writing into a func. There may be one already.